python/tests/test_conversions.py

```diff
@@ -0,0 +1,442 @@
+    """Test round trip conversions of calling `conversions.from_bytes` and then `conversions.to_bytes` on the result"""
+    value_from_bytes = conversions.from_bytes(primitive_type, b)
+    assert value_from_bytes == result
+
+    bytes_from_value = conversions.to_bytes(primitive_type, value_from_bytes)
+    assert bytes_from_value == b
+
+
+@pytest.mark.parametrize(
+    "primitive_type, expected_max_value",
+    [
+        (DecimalType(6, 2), Decimal("9999.99")),
+        (DecimalType(10, 10), Decimal(".9999999999")),
+        (DecimalType(2, 1), Decimal("9.9")),
+        (DecimalType(38, 37), Decimal("9.9999999999999999999999999999999999999")),
+        (DecimalType(20, 1), Decimal("9999999999999999999.9")),
+    ],
+)
+def test_max_value_round_trip_conversion(primitive_type, expected_max_value):
+    """Test round trip conversions of maximum DecimalType values"""
+    b = conversions.to_bytes(primitive_type, expected_max_value)
+    value_from_bytes = conversions.from_bytes(primitive_type, b)
+
+    assert value_from_bytes == expected_max_value
+
+
+@pytest.mark.parametrize(
+    "primitive_type, expected_min_value",
+    [
+        (DecimalType(6, 2), Decimal("-9999.99")),
+        (DecimalType(10, 10), Decimal("-.9999999999")),
+        (DecimalType(2, 1), Decimal("-9.9")),
+        (DecimalType(38, 37), Decimal("-9.9999999999999999999999999999999999999")),
+        (DecimalType(20, 1), Decimal("-9999999999999999999.9")),
+    ],
+)
+def test_min_value_round_trip_conversion(primitive_type, expected_min_value):
+    """Test round trip conversions of minimum DecimalType values"""
+    b = conversions.to_bytes(primitive_type, expected_min_value)
+    value_from_bytes = conversions.from_bytes(primitive_type, b)
+
+    assert value_from_bytes == expected_min_value
+
+
+def test_raise_on_unregistered_type():
+    """Test raising when a conversion is attempted for a type that has no registered method"""
+
+    class FooUnknownType:
+        def __repr__(self):
+            return "FooUnknownType()"
+
+    with pytest.raises(TypeError) as exc_info:
+        conversions.from_partition_value_to_py(FooUnknownType(), "foo")
+    assert (f"Cannot convert partition string to python built-in, type FooUnknownType() not supported: 'foo'") in str(
+        exc_info.value
+    )
+
+    with pytest.raises(TypeError) as exc_info:
+        conversions.to_bytes(FooUnknownType(), "foo")
+    assert ("Cannot serialize value, type FooUnknownType() not supported: 'foo'") in str(exc_info.value)
+
+    with pytest.raises(TypeError) as exc_info:
+        conversions.from_bytes(FooUnknownType(), b"foo")
+    assert ("Cannot deserialize bytes, type FooUnknownType() not supported: b'foo'") in str(exc_info.value)
+
+
+@pytest.mark.parametrize(
+    "primitive_type, value, expected_error_message",
+    [
+        (DecimalType(7, 3), Decimal("123.4567"), "Cannot serialize value, scale of value does not match type decimal(7, 3): 4"),
```

Comment:
We can probably omit a few words from the error message here: "scale does not match {decimal_type}"

Reply:
Fixed!